src/model/orchestrator/services/core/runner-availability-service.ts

```
@@ -0,0 +1,142 @@
+    }
+  }
+
+  /**
+   * Fetch all runners for a repository, handling pagination.
+   */
+  private static async fetchRunners(octokit: Octokit, owner: string, repo: string): Promise<GitHubRunner[]> {
+    const allRunners: GitHubRunner[] = [];
+    let page = 1;
+    const perPage = 100;
+
+    while (true) {
```

ESLint

[error] Unexpected constant condition.

(no-constant-condition)